### ISSUE TITLE
Windows "retina" visual bugs

### ISSUE THREAD

**Issue by mrjbq7** [slavapestov/factor#1405]
I'm running Factor on Windows in a VM on my Mac which has retina screen.

The fonts are pixelated and there are a few minor bugs with positioning.

**Comment by bjourne** [comment]
Does Factor support retina on os x? When I run it on Windows I run it in compatibility mode so everything is scaled up 2x. Turning off compatibility mode makes fonts and other gui elements microscopic.

**Comment by mrjbq7** [comment]
Yeah Factor looks awesome on retina OS X except for a few minor bugs mostly with images not being 2x images.

On Windows VM on OS X (and probably on HiDPI screens) its all pixelated and scaled up.  Thats what I'd like to fix but I haven't looked into what API calls we would need to "retina" it.

**Comment by bpollack** [comment]
I can grab this. There's really two pieces here:

  1. First, `factor.exe` needs to be tagged system-DPI aware, which mostly just involves adding a manifest and a _very_ amount of patching, since most of the hard work was already done to handle Retina on OS X. That'll get Factor to show up as HiDPI when launched on a HiDPI display.
  2. Next, we'll need to support DPI change notifications to handle mixed-resolution display settings (`WM_DPICHANGED`). That'll let Factor sanely handle multi-monitor setups.

I'll try to take a stab at it this evening.

**Comment by bjourne** [comment]
Last I looked at it I got quite far just by doubling all font sizes. Perhaps that's all what's needed? Detect if the screen is high dpi and then make everything twice as big?

**Comment by mrjbq7** [comment]
On the mac we set the ``gl-scale-factor`` to ``2.0``.

**Comment by bpollack** [comment]
Work on this is going on in https://github.com/bpollack/factor/tree/hidpi. I've got a working manifest (though I haven't yet altered the build process) and have added the relevant functions. Will try to get this wrapped up tomorrow.

**Comment by erg** [comment]
I merged @bpollack 's code and added a compile target that shows the DPI. The last steps are fixing the DPI to be correct and then adding the manifest to the .exe and .com files every time.

**Comment by erg** [comment]
There is a ``<dpiAwareness>PerMonitorV2</dpiAwareness>`` for Windows 1607. If we want backwards compatibility there's a dynamic way.

https://docs.microsoft.com/en-us/previous-versions/windows/desktop/legacy/mt846517(v%3Dvs.85)

**Comment by erg** [comment]
Looks like this. All we need to do is fix the small size somehow.

<img width="308" alt="small-listener" src="https://user-images.githubusercontent.com/27430/76148109-ab002480-6068-11ea-98b4-17dc84a7c9c8.png">

**Comment by bpollack** [comment]
Did you make more progress on this @erg? I'm up for giving it another shot this weekend; I'll have some downtime for once. I think we "just" need to double the sizes of everything like the Mac version does.

**Comment by erg** [comment]
If using ``build.cmd`` you need to do the manifest step manually.

From ``Nmakefile``:

```
factor.exe.manifest: factor.exe
	copy factor.exe.manifest.in factor.exe.manifest
	mt -manifest factor.exe.manifest -outputresource:"factor.exe;#1"
```

I get the tiny window but simply setting ``2.0 gl-scale-factor set-global`` doesn't do it.
